app/forms/fields.py

```diff
@@ -163,24 +164,61 @@ def get_integer_field(answer, label, guidance, error_messages):
 
     mandatory_or_optional = get_mandatory_validator(answer, answer_errors)
 
-    if answer['type'] == 'Integer':
-        validate_with = mandatory_or_optional + [
-            IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(maximum=9999999999, messages=answer_errors),
-        ]
-    elif answer['type'] == 'Percentage':
-        validate_with = mandatory_or_optional + [
-            IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(minimum=0, maximum=100, messages=answer_errors),
-        ]
-    else:
-        validate_with = mandatory_or_optional + [
-            IntegerCheck(answer_errors['NOT_INTEGER']),
-            NumberRange(minimum=0, maximum=9999999999, messages=answer_errors),
-        ]
+    max_decimals = answer.get('decimal_places', 0)
+    if max_decimals > MAX_DECIMAL_PLACES:
+        max_decimals = MAX_DECIMAL_PLACES
 
-    return CustomIntegerField(
-        label=label,
-        description=guidance,
-        validators=validate_with,
-    )
+    min_value = get_schema_defined_max_min_value(answer.get('min_value'), answer_store) or 0
+
+    max_value = get_schema_defined_max_min_value(answer.get('max_value'), answer_store) \
+        or get_max_min_system_limits('max', max_decimals)
+
+    if min_value > max_value:
+        raise Exception('min_value: ' + str(min_value) + ' > max_value: ' + str(max_value) +
+                        ' for answer id: ' + answer['id'])
+
+    validate_with = mandatory_or_optional + [
+        NumberCheck(answer_errors['INVALID_NUMBER']),
+        NumberRange(minimum=min_value, maximum=max_value, messages=answer_errors),
+        DecimalPlaces(max_decimals=max_decimals, messages=answer_errors),
+    ]
+
+    if max_decimals > 0:
+        return CustomDecimalField(
+            label=label,
+            validators=validate_with,
+            description=guidance,
+        )
+    else:
+        return CustomIntegerField(
+            label=label,
+            validators=validate_with,
+            description=guidance,
+        )
+
+
+def get_schema_defined_max_min_value(definition, answer_store):
+    if definition:
+        if 'value' in definition:
+            return definition['value']
+        else:
+            answer_id = definition['answer_id']
+            for answer in answer_store.answers:
+                if answer.get('answer_id') == answer_id:
+                    return answer.get('value')
+
+    return None
+
+
+def get_max_min_system_limits(min_or_max, decimals):
+    default = ''
+    for x in range(0, MAX_NUMBER_LENGTH):
+        if x == 0 and min_or_max == 'min':
+            default = '-' + default
+        else:
+            if decimals > 0 and x == MAX_NUMBER_LENGTH - (decimals+1):
+                default = default + '.'
+            else:
+                default = default + '9'
+
+    return int(default) if decimals == 0 else float(default)
```

Comment:
I know we discussed doing this, but seeing it I think we're doing the wrong thing. Using limits based on the 'string length' of a number are downstream limitations. I think any downstream limits should be explicit in the schema. 

For now I think eQ system limits of -9999999999 to 9999999999 and a maximum of 6 decimal places makes sense, and defaults of 0 to 9999999999 with no decimal places. I also think we should throw an exception if we get any values that exceed these (rather than setting to the system maximum)